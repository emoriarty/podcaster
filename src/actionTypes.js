export const FETCH_COUNTRIES = 'itunes/FETCH_COUNTRIES';
export const FETCH_COUNTRIES_SUCCESS = 'itunes/FETCH_COUNTRIES_SUCCESS';
export const FETCH_COUNTRIES_FAIL = 'itunes/FETCH_COUNTRIES_FAIL';

export const FETCH_MEDIA_TYPES = 'itunes/FETCH_MEDIA_TYPES';
export const FETCH_MEDIA_TYPES_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_SUCCESS';
export const FETCH_MEDIA_TYPES_FAIL = 'itunes/FETCH_MEDIA_TYPES_FAIL';

export const FETCH_COMMON_TRANSLATIONS = 'itunes/FETCH_COMMON_TRANSLATIONS';
export const FETCH_COMMON_TRANSLATIONS_SUCCESS = 'itunes/FETCH_COMMON_TRANSLATIONS_SUCCESS';
export const FETCH_COMMON_TRANSLATIONS_FAIL = 'itunes/FETCH_COMMON_TRANSLATIONS_FAIL';

export const FETCH_MEDIA_TYPES_TRANSLATIONS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL';
