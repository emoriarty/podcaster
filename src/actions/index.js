export { default as fetchCountries } from './itunes/fetchCountries';
export { default as fetchMediaTypes } from './itunes/fetchMediaTypes';
