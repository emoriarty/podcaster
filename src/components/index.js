export * from './Expansion'
export * from './Layouts'
export * from './ListItems'
export * from './Texts'
