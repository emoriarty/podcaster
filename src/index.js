import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import configStore from './store'
import injectTapEventPlugin from 'react-tap-event-plugin'
import App from './containers/App'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import '../styles/main.scss'

injectTapEventPlugin()

const store = configStore()

ReactDOM.render(
  <Provider store={store}>
    <MuiThemeProvider>
      <App />
    </MuiThemeProvider>
  </Provider>,
  document.getElementById('app')
)
