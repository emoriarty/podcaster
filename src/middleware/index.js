export { default as yqlMiddleware } from './yql'
export { default as rssMiddleware } from './rss'
