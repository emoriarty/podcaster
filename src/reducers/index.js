import { combineReducers } from 'redux'
import countries from './countries'
import entities from './entities'
import mediaTypes from './mediaTypes'
import settings from './settings'
import translations from './translations'
import podcasts from './podcasts'

const app = combineReducers({
  entities,
  countries,
  mediaTypes,
  settings,
  translations,
  podcasts
})

export default app
