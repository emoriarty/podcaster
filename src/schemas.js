import { schema } from 'normalizr'

export const country = new schema.Entity(
  'countries',
  {},
  { idAttribute: 'country_code' }
)

export const mediaType = new schema.Entity('mediaTypes')

export const podcast = new schema.Entity(
  'podcasts',
  {},
  {
    idAttribute: (value, parent, key) => {
      return value.id.attributes['im:id']
    }
  })

export const podcastFeed = { feed: { entry: new schema.Array(podcast) } }
