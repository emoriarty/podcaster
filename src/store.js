import { createStore, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { yqlMiddleware } from './middleware'
import appReducer from './reducers'

const loggerMiddleware = createLogger()

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

export default () =>
  createStore(
    appReducer,
    composeEnhancers(
      applyMiddleware(
        thunkMiddleware,
        yqlMiddleware,
        loggerMiddleware
      )
    )
  )
