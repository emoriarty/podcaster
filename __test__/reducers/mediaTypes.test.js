import reducer from '../../src/reducers/mediaTypes'
import * as types from '../../src/actionTypes'
import mockResponse from '../../src/utils/__mocks__/media-types'

describe('mediaTypes reducer', () => {
  it('should return the initial state', () =>
    expect(reducer(undefined, {}))
      .toMatchSnapshot()
  )

  it('when action FETCH_MEDIA_TYPES sets isFecthing property to true', () => {
    const action = {
      type: types.FETCH_MEDIA_TYPES
    }
    const next = reducer(undefined, action)
    expect(next.isFetching)
      .toBeTruthy()
  })

  it('should handle FETCH_MEDIA_TYPES_SUCCESS', () => {
    const action = {
      type: types.FETCH_MEDIA_TYPES_SUCCESS,
      payload: { result: mockResponse.query.results.json.json }
    }
    expect(reducer([], action))
      .toMatchSnapshot()
  })

  it('should handle FETCH_MEDIA_TYPES_FAIL', () => {
    const action = {
      type: types.FETCH_MEDIA_TYPES_FAIL,
      error: new Error()
    }
    expect(reducer([], action))
      .toMatchSnapshot()
  })
})
