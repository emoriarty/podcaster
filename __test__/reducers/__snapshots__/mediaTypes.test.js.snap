// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mediaTypes reducer should handle FETCH_MEDIA_TYPES_FAIL 1`] = `
Object {
  "error": [Error],
  "isFetching": false,
}
`;

exports[`mediaTypes reducer should handle FETCH_MEDIA_TYPES_SUCCESS 1`] = `
Object {
  "ids": Array [
    Object {
      "canBeExplicit": true,
      "feed_types": Array [
        Object {
          "translation_key": "topaudiobooks",
          "urlPrefix": "https://itunes.apple.com/<%= country_code %>/rss/topaudiobooks/<%= parameters %>",
          "urlSuffix": "xml",
        },
      ],
      "id": "50000024",
      "store": "music",
      "subgenres": Array [
        Object {
          "id": "",
          "translation_key": "all",
        },
        Object {
          "id": "50000041",
          "translation_key": "arts&entertainment",
        },
      ],
      "translation_key": "audiobooks",
    },
  ],
  "isFetching": false,
}
`;

exports[`mediaTypes reducer should return the initial state 1`] = `
Object {
  "error": null,
  "ids": Array [],
  "isFetching": false,
}
`;
